coalib/results/result_actions/ShowPatchAction.py

```diff
@@ -104,9 +104,10 @@ def apply(self,
             from coalib.output.ConsoleInteraction import print_result
             # Most of the params are empty because they're unneeded in
             # noninteractive mode. Yes, this cries for a refactoring...
-            print_result(printer, None, {}, result, {}, interactive=False)
```

Comment:
I need help with this part, here result object is required as a parameter to `print_result` function but now we don't have result object to pass.

Reply:
Modify the `print_result` function then. Idk which is the best way of doing this, but you can directly pass the data needed to print, or maybe creating a new class to hold the data (this seems wrong since we already have a Result class for that), or passing a mocked result object.

Have you considered passing Result object to the actions? What are the downsides of doing that?

Reply:
Currently, Result objects are passed to actions, but now Result has an attribute `actions` which is a list of actions. How can I pass Result to actions which are inside Result itself?

Reply:
Not passing the same Reuslt object, but creating a new one, in the new object only the data required is provided and everything else will be ignored. This would just 'work', and I don't think this is the way we want to implement. But still you can try it out to test.
Another approach as I mentioned was to create a new class similar to Result and changing the print and other functions as per that (i.e. wherever the result was passed).